### PR TITLE
Bump k8s dependencies to 1.27.7

### PR DESCRIPTION
**Pull request by jlojosnegros**

### CODE REVIEW

File: test/e2e/sched/sched_test.go [hunk 1]
```diff
@@ -308,3 +309,16 @@ func findKubeSchedulerProfileByName(sc *schedconfig.KubeSchedulerConfiguration,
 
 	return nil, nil
 }
+
+func schedulerConfigurationFromData(data []byte) (*schedconfig.KubeSchedulerConfiguration, error) {
```
Comment (ffromani) [inline test/e2e/sched/sched_test.go:313]: let's use `manifests.DecodeSchedulerProfilesFromData`, and remove the dep on the scheduler config apis (which transitively pull `k8s.io/kubernetes`)

**Review by ffromani — COMMENTED** [review]
thanks, it's a nice start. Unfortunately we severely lagged behind both `rte` and `deployer` and we need to make sure we are using the new APIs correctly, and that we deal with changes like https://github.com/k8stopologyawareschedwg/deployer/pull/207 https://github.com/k8stopologyawareschedwg/deployer/pull/198 https://github.com/k8stopologyawareschedwg/deployer/pull/199 https://github.com/k8stopologyawareschedwg/deployer/pull/184 https://github.com/k8stopologyawareschedwg/deployer/pull/186

**Comment by ffromani** [comment]
bogus retest (I was looking at the wrong PR)

File: controllers/numaresourcesscheduler_controller_test.go [hunk 2]
```diff
@@ -405,11 +406,13 @@ func pop(m map[string]string, k string) string {
 }
 
 func diffYAML(want, got string) (string, error) {
```
[the remark this answers is not in the document]
Comment (jlojosnegros) [reply controllers/numaresourcesscheduler_controller_test.go]: changed

File: internal/wait/daemonset.go [hunk 3]
```diff
@@ -26,8 +26,10 @@ import (
 
 func (wt Waiter) ForDaemonSetReadyByKey(ctx context.Context, key ObjectKey) (*appsv1.DaemonSet, error) {
 	updatedDs := &appsv1.DaemonSet{}
-	err := k8swait.PollImmediate(wt.PollInterval, wt.PollTimeout, func() (bool, error) {
-		err := wt.Cli.Get(ctx, key.AsKey(), updatedDs)
+
+	immediate := true
+	err := k8swait.PollUntilContextTimeout(ctx, wt.PollInterval, wt.PollTimeout, immediate, func(aContext context.Context) (bool, error) {
```
Comment (ffromani) [inline internal/wait/daemonset.go:31]: was the linter complaining about those?

Comment (jlojosnegros) [reply internal/wait/daemonset.go:31]: yeah It seems that `PollImmediate` is also deprecated.

**Comment by ffromani** [comment]
fixes: #677

**Pull request closed without merging**